modules/system/classes/FileManifest.php

```diff
@@ -0,0 +1,171 @@
+<?php namespace System\Classes;
+
+use ApplicationException;
+use Config;
+use October\Rain\Filesystem\Filesystem;
+use October\Rain\Halcyon\Datasource\FileDatasource;
+
+/**
+ * Stores the file manifest for this October CMS installation.
+ *
+ * This manifest is a file checksum of all files within this October CMS installation. When compared to the source
+ * manifest, this allows us to determine the current installation's build number.
+ *
+ * @package october\system
+ * @author Ben Thomson
+ */
+class FileManifest
+{
+    /**
+     * @var string Root folder of this installation.
+     */
+    protected $root;
+
+    /**
+     * @var string Modules to store in manifest.
+     */
+    protected $modules = ['system', 'backend', 'cms'];
+
+    /**
+     * @var array Files cache.
+     */
+    protected $files = [];
+
+    /**
+     * Constructor.
+     *
+     * @param string $root The root folder to get the file list from.
+     * @param array $modules An array of modules to include in the file manifest.
+     */
+    public function __construct($root = null, array $modules = null)
+    {
+        if (isset($root)) {
+            $this->setRoot($root);
+        } else {
+            $this->setRoot(base_path());
+        }
+
+        if (isset($modules)) {
+            $this->setModules($modules);
+        } else {
+            $this->setModules(Config::get('cms.loadModules', ['System', 'Backend', 'Cms']));
+        }
+    }
+
+    /**
+     * Sets the root folder.
+     *
+     * @param string $root
+     * @throws ApplicationException If the specified root does not exist.
+     */
+    public function setRoot($root)
+    {
+        if (is_string($root)) {
+            $this->root = realpath($root);
+
+            if ($this->root === false || !is_dir($this->root)) {
+                throw new ApplicationException(
+                    'Invalid root specified for the file manifest.'
+                );
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * Sets the modules.
+     *
+     * @param array $modules
+     */
+    public function setModules(array $modules)
+    {
+        $this->modules = array_map(function ($module) {
+            return strtolower($module);
+        }, $modules);
+
+        return $this;
+    }
+
+    /**
+     * Gets a list of files and their corresponding hashsums.
+     *
+     * @return array
+     */
+    public function getFiles()
+    {
+        if (count($this->files)) {
+            return $this->files;
+        }
+
+        $files = [];
+
+        foreach ($this->modules as $module) {
+            $path = $this->root . DIRECTORY_SEPARATOR . 'modules' . DIRECTORY_SEPARATOR . $module;
```

Comment:
Since DIRECTORY_SEPERATOR doesn't actually provide any real-world benefit (see https://dev.to/c33s/always-use--as-directory-seperator-in-php-43l7) I'd prefer that we just standardized on forward slashes in paths always because then it's easier to read and more consistent.

Reply:
@LukeTowers AFAIK, Laravel still use it pretty extensively in file handling. I gotta assume there's a reason for that.

I read the post you linked but that seemed to only make the argument that these paths *might* end up in URLs where they will encode incorrectly. That wouldn't occur with this command.

Reply:
@bennothommo i tested it and it doesnt work in windows looks like

Reply:
@Samuell1 Interesting - what does the path look like on Windows with `DIRECTORY_SEPARATOR`?

Reply:
@Samuell1 @LukeTowers FileDatasource just uses forward-slashes, so I'll remove the `DIRECTORY_SEPARATOR` constant. If it becomes a problem, we'll revisit it.